ert_shared/libres_facade.py

```diff
@@ -192,20 +200,22 @@ def refcase_data(self, key):
 
         return data.iloc[1:]
 
-    def history_data(self, key, case=None):
+    def history_data(self, key, case=None, read_only=False):
         if not self.is_summary_key(key):
             return DataFrame()
 
         if case is None:
             return self.refcase_data(key)
 
-        data = self.gather_summary_data(case, key)
+        data = self.gather_summary_data(case, key, read_only=read_only)
         if data.empty and case is not None:
             data = self.refcase_data(key)
 
         return data
 
-    def gather_gen_data_data(self, case, key, realization_index=None):
+    def gather_gen_data_data(
+        self, case, key, realization_index=None, read_only: bool = False
+    ):
         """:rtype: pandas.DataFrame"""
```

Comment:
`:rtype:` is deprecated in favor of type-hints and the docstring needs to be updated.